File: src/test/java/akkastudy/deathwatch/java/DeathWatchTest.java
package akkastudy.deathwatch.java;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import akkastudy.SenderActor;
import org.junit.Test;

/**
 * User: Daniel Hinojosa (dhinojosa@evolutionnext.com)
 * Date: 3/9/13
 * Time: 12:18 PM
 */
public class DeathWatchTest {

    @Test
    public void testDeathWatch() throws InterruptedException {
        ActorSystem system = ActorSystem.create("MySystem");
        ActorRef actor = system.actorOf(Props.create(DeathWatchActor.class), "deathWatcher");
        ActorRef sender = system.actorOf(Props.create(SenderActor.class), "senderActor");
        Thread.sleep(3000);
        actor.tell("kill", sender);
        Thread.sleep(3000);
    }
}

File: src/test/scala/akkastudy/deathwatch/scala/DeathWatchTest.scala
package akkastudy.deathwatch.scala

import org.scalatest.{Matchers, FlatSpec}
import akka.actor.{Props, ActorSystem}

/**
 *
 * @author Daniel Hinojosa
 * @since 3/26/14 12:09 PM
 *        url: <a href="http://www.evolutionnext.com">http://www.evolutionnext.com</a>
 *        email: <a href="mailto:dhinojosa@evolutionnext.com">dhinojosa@evolutionnext.com</a>
 *        tel: 505.363.5832
 */
class DeathWatchTest extends FlatSpec with Matchers {
  behavior of "Death Watch"

  it should "An actor can be notified of a termination by using the same supervisory facilities" in {
    val actorSystem = ActorSystem("DeathWatchActorSystem")
    val deathWatcherActor = actorSystem.actorOf(Props[DeathWatcherActor], "DeathWatchActor")
    deathWatcherActor ! "kill"
  }
}
